File: Python/pyramid.py
"""
|--------------------------------------------------------------------------
| Pyramid Array
|--------------------------------------------------------------------------
|
"""

def pyramid(n):
   res = []
   for i in range(n):
      temp = []
      for j in range(i+1):
         temp.append(1)
      res.append(temp)
   return res


pyramid = lambda n:[-~i*[1]for i in range(n)]


def pyramid(n):
   return [[1]*x for x in range(1, n+1)]


pyramid = lambda n:[i*[1]for i in range(1,n+1)]


def pyramid(n):
   result = []
   for i in range (1, n + 1):
      result.append(i * [1])
   return result


print(pyramid(0), [])
print(pyramid(1), [[1]])
print(pyramid(2), [[1], [1, 1]])
print(pyramid(3), [[1], [1, 1], [1, 1, 1]])

File: PHP/6php 1/pyramid.php
<?php
/*
|--------------------------------------------------------------------------
| Pyramid Array
|--------------------------------------------------------------------------
|
*/

function pyramid($n) {
   $res = [];
   for ($i = 0; $i < $n; $i++) {
      $temp = [];
      for ($j = 0; $j <= $i; $j++) {
         array_push($temp, 1);
      }
      array_push($res, $temp);
   }
   return $res;
}


function pyramid($n) {
   return $n? array_map(function($i) {return array_fill(0, $i, 1);}, range(1, $n)): [];
}


function pyramid($n) {
   $out = [];
   if ($n > 0) {
      $u=0;
      do {
         array_push($out, array_fill(0, ++$u, 1));
      } while (--$n > 0);
   }
   return $out;
}


function pyramid($n) {
   $res = [];
   for ($i = 1; $i <= $n; $i++) {
      $res[] = array_fill(0, $i, 1);
   }
   return $res;
}


print_r(pyramid(0));
print_r(pyramid(2));


// []
// [ [1], [1, 1] ]